src/Features/Blockcore.Features.Wallet/MultisigManager.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Blockcore.Consensus;
+using Blockcore.Consensus.ScriptInfo;
+using Blockcore.Consensus.TransactionInfo;
+using Blockcore.Features.Wallet.Interfaces;
+using Blockcore.Features.Wallet.Types;
+using Blockcore.Interfaces;
+using Blockcore.Networks;
+using NBitcoin;
+
+namespace Blockcore.Features.Wallet
+{
+    public class MultisigManager : IMultisigManager
+    {
+        private readonly Network _network;
+        private readonly IWalletTransactionHandler _walletTransactionHandler;
+        private readonly IWalletManager _walletManager;
+        private readonly IBroadcasterManager _broadcasterManager;
+        private readonly IConsensusManager _consensusManager;
+        private readonly IBlockStore _blockStore;
+
+        public MultisigManager(Network network, IWalletTransactionHandler walletTransactionHandler, IWalletManager walletManager, IBroadcasterManager broadcasterManager, IConsensusManager consensusManager, IBlockStore blockStore)
+        {
+            this._network = network;
+            this._walletTransactionHandler = walletTransactionHandler;
+            this._walletManager = walletManager;
+            this._broadcasterManager = broadcasterManager;
+            this._consensusManager = consensusManager;
+            this._blockStore = blockStore;
+        }
+
+        public async Task<string> FundMultisigAsync(FundMultisigInput input)
+        {
+
+            var listOfPubKeys =
+                input
+                .PublicKeys
+                .Select(l => new PubKey(l))
+                .ToList();
+
+            // require minimum of 50% consensus
+            var minimumSignatories = (listOfPubKeys.Count / 2) + 1;
```

Comment:
Do you need the + 1? With this logic, three public signers, then it will only require 3.5 signers.

Reply:
Since I am dividing integers the result is just the integer part. So 5/2 = 2, then add one its 3 of 5

Reply:
Thank you for the clarity. You could use Decimal.Divide with listOfPubKeys.Count for more precision?

Reply:
Nevermind, I see sigCount is an int. This works.